src/extension.ts

```diff
@@ -260,6 +289,53 @@ export class Extension implements vscode.Disposable {
     return this.enabled = enabled
   }
 
+  /**
+   * Make all selections in the editor non-empty by selecting at least one character.
+   */
+  normalizeSelections(editor: vscode.TextEditor) {
+    if (this.allowEmptySelections || this.ignoreSelectionChanges)
+      return
+
+    if (this.modeMap.get(editor.document) !== Mode.Normal)
+      return
+
+    // Since this is called every time when selection changes, avoid allocations
+    // unless really needed and iterate manually without using helper functions.
+    let normalizedSelections
+
+    for (let i = 0; i < editor.selections.length; i++) {
+      const selection = editor.selections[i]
+      if (selection.isEmpty) {
+        if (!normalizedSelections) {
+          // Change needed. Allocate the new array and copy what we have so far.
+          normalizedSelections = editor.selections.slice(0, i)
+        }
+
+        const active = selection.active
+
+        if (active.character >= editor.document.lineAt(active.line).range.end.character) {
+          // Selection is at line end. Just keep it that way.
+          normalizedSelections.push(selection)
+        } else {
+          const offset = editor.document.offsetAt(selection.active)
+          const nextPos = editor.document.positionAt(offset + 1)
+
+          if (nextPos.isAfter(selection.active)) {
+            // Move anchor to select 1 character after, but keep the cursor position.
+            normalizedSelections.push(new vscode.Selection(active.translate(0, 1), active))
+          } else {
+            // Selection is at the very end of the document. Select the last character instead.
+            normalizedSelections.push(new vscode.Selection(active, active.translate(0, -1)))
+          }
+        }
+      } else if (normalizedSelections)
```

Comment:
Please wrap the `else if` block in brackets.
```suggestion
      } else if (normalizedSelections !== undefined) {
```